src/repositories/versionable/VersionableRepository.ts

```diff
@@ -11,6 +12,9 @@ export default class VersionableRepository<D extends mongoose.Document, M extend
 
     protected create(data: any): Promise<D> {
         console.log('UserRepository:: create', data);
+        const salt = bcrypt.genSaltSync(10);
+         const hash = bcrypt.hashSync(data.password, salt);
+        data.password = hash;
```

Comment:
Not put hashing logic here, Since it's a generic class and can be used to create a repository for other models that do not have a password field. Put this in UserRepository or in User Conrolller.